Changed TF Export Backend to Keras and added support for Eltwise and BatchNorm layers

PalashTanejaPro opened the pull request:
Tensorflow export used caffe-tensorflow as a major dependency, but the project has a lot of bugs.

Our previous export did not work for a majority of the Fabrik Zoo models, but this one has worked with all of the models in my testing.

To fix this problem I used our own Keras export to generate a .json and created a function that converts this .json into the Tensorflow supported .pbtxt . 

This removes our dependency on caffe-tensorflow and might even remove our dependency on caffe if we remove a few checks.

utsavgarg commented on tensorflow_app/views/export_graphdef.py:
```diff
@@ -1,58 +1,196 @@
 
+def json2pbtxt(json_string, f, output_fld):
+    net_model = model_from_json(json_string)
+    num_output = 1
```
Why create this variable if it is hardcoded?

PalashTanejaPro replied:
Yeah, this was carried over from my script.

utsavgarg commented on tensorflow_app/views/export_graphdef.py:
```diff
@@ -1,58 +1,196 @@
 
 
+def json2pbtxt(json_string, f, output_fld):
+    net_model = model_from_json(json_string)
```
Could we have directly passed the model instead of first converting to JSON and then back to model?

PalashTanejaPro replied:
The user needs to run it using tensorflow backend

utsavgarg commented on tensorflow_app/views/export_graphdef.py:
```diff
@@ -1,58 +1,196 @@
+            final_output.append(net_out[j])
+
+        model = Model(inputs=final_input, outputs=final_output, name=net_name)
+        json_string = Model.to_json(model)
```
It would be better if we can reuse the code from `export_json` instead of maintaining it in two places.

PalashTanejaPro replied:
That would need a minor change at both the places

PalashTanejaPro commented:
@utsavgarg 
- Added a new script that converts json2pbtxt so that we can use any backend while running.
- Now export tf directly imports the code from the keras app
- Removed hard coded variables and the json2pbtxt function which would have worked only on TF backend of Keras.
